providers/inwx/inwxProvider.go

```diff
@@ -0,0 +1,292 @@
+	/* INWX is a little bit special for CNAME,NS,MX and SRV records:
+	   The API will not accept any target with a final dot but will
+	   instead always add this final dot internally.
+	*/
+	case "CNAME", "NS":
+		req.Content = targetWithoutDot
+	case "MX":
+		req.Priority = int(rec.MxPreference)
+		req.Content = targetWithoutDot
+	case "SRV":
+		req.Priority = int(rec.SrvPriority)
+		req.Content = fmt.Sprintf("%d %d %v", rec.SrvWeight, rec.SrvPort, targetWithoutDot)
+	default:
+		req.Content = rec.GetTargetCombined()
+	}
+
+	return req
+}
+
+func (api *InwxApi) createRecord(domain string, rec *models.RecordConfig) error {
+	req := makeNameserverRecordRequest(domain, rec)
+	_, err := api.client.Nameservers.CreateRecord(req)
+	return err
+}
+
+func (api *InwxApi) updateRecord(RoId int, rec *models.RecordConfig) error {
+	req := makeNameserverRecordRequest("", rec)
+	err := api.client.Nameservers.UpdateRecord(RoId, req)
+	return err
+}
+
+func (api *InwxApi) deleteRecord(RoId int) error {
+	return api.client.Nameservers.DeleteRecord(RoId)
+}
+
+func (api *InwxApi) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	dc.Punycode()
+
+	foundRecords, err := api.GetZoneRecords(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	differ := diff.New(dc)
+	_, create, del, mod := differ.IncrementalDiff(foundRecords)
+	corrections := []*models.Correction{}
+
+	for _, d := range create {
+		des := d.Desired
+		corrections = append(corrections, &models.Correction{
+			Msg: d.String(),
+			F:   func() error { return api.createRecord(dc.Name, des) },
+		})
+	}
+	for _, d := range del {
+		existingId := d.Existing.Original.(goinwx.NameserverRecord).Id
+		corrections = append(corrections, &models.Correction{
+			Msg: d.String(),
+			F:   func() error { return api.deleteRecord(existingId) },
+		})
+	}
+	for _, d := range mod {
+		rec := d.Desired
+		existingId := d.Existing.Original.(goinwx.NameserverRecord).Id
+		corrections = append(corrections, &models.Correction{
+			Msg: d.String(),
+			F:   func() error { return api.updateRecord(existingId, rec) },
+		})
+	}
+
+	return corrections, nil
+}
+
+func (api *InwxApi) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	if api.sandbox {
+		return models.ToNameservers(InwxSandboxDefaultNs)
+	} else {
+		return models.ToNameservers(InwxDefaultNs)
+	}
+}
+
+func (api *InwxApi) GetZoneRecords(domain string) (models.Records, error) {
+	info, err := api.client.Nameservers.Info(domain, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	var records = []*models.RecordConfig{}
+
+	for _, record := range info.Records {
+		if record.Type == "SOA" {
+			continue
+		}
+
+		/* INWX is a little bit special for CNAME,NS,MX and SRV records:
+		   The API will not accept any target with a final dot but will
+		   instead always add this final dot internally.
+		*/
+		if record.Type == "CNAME" || record.Type == "MX" || record.Type == "NS" || record.Type == "SRV" {
+			if record.Content != "." {
```

Comment:
The comment doesn't mention in what situation where the name is "."

Reply:
Good catch, I forgot this code from an early version where I was trying to figure out why these records didn't work.